fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/RemoveAggregationFromAggTable.java

```diff
@@ -129,19 +127,30 @@ public List<OptExpression> transform(OptExpression input, OptimizerContext conte
             // in this case, CallOperator must have only one child ColumnRefOperator
             projectMap.put(entry.getKey(), entry.getValue().getChild(0));
         }
+
+        Map<ColumnRefOperator, ScalarOperator> newProjectMap = Maps.newHashMap();
+        ReplaceColumnRefRewriter rewriter = new ReplaceColumnRefRewriter(projectMap);
+
+        OptExpression newChildOpt = input.inputAt(0);
+        if (aggregationOperator.getPredicate() != null) {
+            // rewrite the having predicate. replace the aggFunc by the columnRef
+            ScalarOperator newPredicate = rewriter.rewrite(aggregationOperator.getPredicate());
+            LogicalFilterOperator filterOperator = new LogicalFilterOperator(newPredicate);
+            newChildOpt = OptExpression.create(filterOperator, newChildOpt);
+        }
```

Comment:
BTW,  we can add the predicate on scan direct? because the predicate pushdown has already ended when this rule executed

Reply:
Good idea.